ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSshell.cpp

```diff
@@ -89,14 +89,21 @@ string execute_shell_for_output(const string &command) {
     CloseHandle(pi.hThread);
     CloseHandle(hStdOutPipeWrite);
     CloseHandle(hStdInPipeRead);
-    for (;;) {
+    HANDLE waitHandles[] = { pi.hProcess, hStdOutPipeRead };
+    while (DWORD eventSignalId = MsgWaitForMultipleObjects(2, waitHandles, false, INFINITE, QS_ALLEVENTS)) {
+      if (eventSignalId == WAIT_OBJECT_0) break;
+      MSG msg;
+      while (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE)) {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+      }
```

Comment:
Replace this with `enigma::handleEvents();` please, and you might need to include `platforms_mandatory.h` then.

Reply:
Alrighty then. Bam! I have done it. :)

Reply:
You have a conflict to resolve on `execute_program` now since I just merged that into master.

Reply:
just saw that; it's now resolved.